### DISCUSSION TITLE
header based routing in canary deployment

### DISCUSSION THREAD

**Discussion by Yogeshbaskaran** [argoproj/argo-rollouts#2716] [Q&A]
Hi Team, 

we are using canary deployment using header based routing for preview testing of dependent services.
With the default functionality from Argo Rollouts, the independent calls to individual services are routed correctly using headers.
However service to service call is not routed to the preview version

Below are the code : (serviceb application and for servicea application only the service will be changed {{ include "servicea.fullname" . }} in both preview and stable ) 

  strategy:
   canary:
    canaryService:  {{ include "serviceb.fullname" . }}-preview
    stableService:  {{ include "serviceb.fullname" . }}
    trafficRouting:
      nginx:
        stableIngress: {{ include "serviceb.fullname" . }}
        additionalIngressAnnotations:   # optional
          canary-by-header: "preview"
          canary-by-header-value: "true"
    steps:
    - setWeight: 0
    - pause: { duration: 40s }
    - setCanaryScale:
        weight: 50
     - pause: {}

Please correct me if am wrong in the above code in header based routing,  also i need your suggestions on what is the best practice when we do canary deployment for dependent service which should have preview and stable service&ingress. 

Share any link or document if have anything which will fix this issue which will be much appreciated.

**Comment by meeech** [comment]
I think if you go service to service, you are bypassing ingress, which means header routing won't work - since that is ingress functionality.
